src/app/introduction/activity-3/activity-3.component.ts

```diff
@@ -19,7 +20,10 @@ export class PicturePuzzleComponent implements OnInit {
     // "#video" is the name of the template variable in the video element
     @ViewChild('video') video: any;
 
-    private PUZZLE_DIFFICULTY = 4;
+    private PUZZLE_DIFFICULTY: number;
+    private difficultyList = [2, 4, 6];
```

Comment:
will this be changed ? if not make it constant. Same with levels.

Reply:
And also include this level list in some common file. Do the same for all activities

Reply:
Done. for making **readonly**

For storing in some common file, may not work here. Because, the way levels are increased in different activities is different. That is sometimes, it is just changing no. of fragments like in this case and then somtimes, it involves fetching a different property from a JSON file like objects marked under 'level1', 'level2' etc. Ex. for different set of questions for different levels.
So having a **readonly** activity-specific declaration at the top, may not be a bad idea.